src/ServicePulse.Host/app/modules/configuration/views/notificationsemailmodal.html

```diff
@@ -0,0 +1,56 @@
+﻿<div class="modal-header">
+    <h3 class="modal-title">Email configuration</h3>
+</div>
+<form name="notificationsForm" novalidate>
+    <div class="modal-body">
+        <div class="row">
+            <div class="form-group" ng-class="{'has-error': notificationsForm.smtpServerAddress.$invalid}">
+                <label for="smtpServerAddress">SMTP server address</label>
+                <input type="text" id="smtpServerAddress" name="smtpServerAddress"
+                    ng-model="settings.smtp_server"
+                    class="form-control" required />
+            </div>
+            <div class="row"></div>
+            <div class="form-group" ng-class="{'has-error': notificationsForm.smtpServerPort.$invalid}">
+                <label for="smtpServerPort">SMTP server port</label>
+                <input type="text" id="smtpServerPort" name="smtpServerPort"
+                    ng-model="settings.smtp_port"
+                    class="form-control" required />
+            </div>
+            <div class="row"></div>
+            <div class="form-group">
+                <label for="account">Authentication account</label>
+                <input type="text" id="account" name="account"
+                    ng-model="settings.authentication_account" class="form-control" />
+            </div>
+            <div class="row"></div>
+            <div class="form-group">
+                <label for="account">Authentication password</label>
+                <input type="password" id="password" name="password"
+                    ng-model="settings.authentication_password" class="form-control" />
+            </div>
+            <div class="row"></div>
+            <div class="form-group">
+                <input type="checkbox" id="enableSSL" name="enableSSL"
+                    ng-model="settings.enable_ssl" class="check-label" />
+                <label for="enableSSL">Use SSL</label>
+            </div>
+            <div class="row"></div>
+            <div class="form-group">
+                <label for="from">From address</label>
+                <input type="from" id="from" name="from"
+                    ng-model="settings.from" class="form-control" />
+            </div>
+            <div class="row"></div>
+            <div class="form-group">
+                <label for="to">To address</label>
+                <input type="to" id="to" name="to"
```

Comment:
Is this validated or is this one of the https://mentalized.net/journal/2019/08/02/email-validation-is-hard/?

Reply:
I've added browser-based validation to both email fields and the port fields